### ISSUE TITLE
OktaClient.Users.GetUserAsync(userId) throws error

### ISSUE THREAD

**Issue by djlaney3** [okta/okta-sdk-dotnet#266]
Using the OktaClient.Users.GetUserAsync(userId) returns the following error:

`{Newtonsoft.Json.JsonReaderException: Unexpected character encountered while parsing value: <. Path '', line 0, position 0.
   at Newtonsoft.Json.JsonTextReader.ParseValue()
   at Newtonsoft.Json.JsonTextReader.Read()
   at Newtonsoft.Json.JsonReader.ReadForType(JsonContract contract, Boolean hasConverter)
   at Newtonsoft.Json.Serialization.JsonSerializerInternalReader.Deserialize(JsonReader reader, Type objectType, Boolean checkAdditionalContent)
   at Newtonsoft.Json.JsonSerializer.DeserializeInternal(JsonReader reader, Type objectType)
   at Newtonsoft.Json.JsonSerializer.Deserialize[T](JsonReader reader)
   at Okta.Sdk.Internal.DefaultSerializer.Deserialize[T](String input)
   at Okta.Sdk.Internal.DefaultDataStore.EnsureResponseSuccess(HttpResponse`1 response)}`

**Comment by nbarbettini** [comment]
Can you share more of your code? It looks like the API is responding with an HTML document instead of a JSON document, but I'm not sure why that would happen.

**Comment by djlaney3** [comment]
Sure.  I'm using .net core 2.1
I inject the service in startup.
` //add okta service
            services.AddSingleton<IOktaClient>(new OktaClient(new OktaClientConfiguration
            {
                OktaDomain = "https://xxxxxx.oktapreview.com/oauth2/default",
                Token = "xxxxxxxxx"
            }));`
Then I pass in an authorized getAccessToken from Vue and get the UID claim in .net and use it like:
`Okta.Sdk.IUser oktaUser = await _oktaClient.Users.GetUserAsync(UID);`
That is when the exception gets thrown.

**Comment by djlaney3** [comment]
I used fiddler to get the url path:
`/oauth2/default/api/v1/users/XXXXXXX HTTP/1.1`

**Comment by djlaney3** [comment]
Now I receive invalid token.  Why would the token not work?
________________________________

**Comment by nbarbettini** [comment]
How did you get the token?

**Comment by nbarbettini** [comment]
Oh sorry, I missed this part:

> Then I pass in an authorized getAccessToken from Vue

Are you signing a user in in your Vue app, and then making an API request to your backend to look up the user in Okta? If I understand your goal a little better, I can suggest a solution.

**Comment by djlaney3** [comment]
Yes, thats exactley what I am doing. I use vue to authenticate dirctly with okta, then authorize in .net webapi, and use okta.sdk to look up user information.

**Comment by nbarbettini** [comment]
Makes sense. You need to make a few changes to your app. Here's what should happen:

1. Your frontend gets an access token from Okta (this part you have done already)
2. Your frontend makes a request to your Web API, like `POST /api/someuseraction`. Attach the access token to the outgoing request in the HTTP Authorization header, like `Authorization: Bearer {token}`. There's an example here: https://github.com/okta/okta-oidc-js/tree/master/packages/okta-vue#use-the-access-token
3. Configure ASP.NET Core to look for that access token and allow the request if the token is valid. The easiest way is with the middleware we just published: https://github.com/okta/okta-aspnet#using-okta-with-aspnet-core-web-api This allows you to have communication between your Vue app and Web API (secured by the access token).
5. To look up the full user profile in your Web API code, configure this .NET SDK with your Okta domain and an [API token](https://developer.okta.com/docs/api/getting_started/getting_a_token) (different than the access token!). Then you can use `GetUserAsync` and look up the uid of the user as you were trying to do above.

Tl;dr - you're getting an error because you're using the wrong type of token with the Okta API.

**Comment by djlaney3** [comment]
Thanks.  All i was missing was the API token.  Case closed.
